### PR TITLE
Force a sync before serializing a non-cached file

### PR DESCRIPTION
**Pull request by jleclanche**

### CODE REVIEW

File: pootle/apps/pootle_store/models.py [hunk 1]
```diff
@@ -1902,6 +1902,7 @@ def serialize(self):
 
         ret = cache.get(path, version=rev)
         if not ret:
+            self.sync()
```
Comment (unho) [inline pootle/apps/pootle_store/models.py]: Is it not better to sync after updating file header?

**Comment by unho** [comment]
Why is the sync necessary?

**Comment by unho** [comment]
Does this address any specific issue? I wonder if #3746 is related to this.

**Comment by dwaynebailey** [comment]
This is completely the wrong way to do this.  Its essentially doing a ``sync_stores`` which will mess with the actual operation of ``sync_stores``.

**Comment by jleclanche** [comment]
Updated to a different approach. @dwaynebailey @julen 1 gtm to land please.

**Comment by unho** [comment]
lgtm

**Comment by julen** [comment]
:+1: fine to merge, thank you @jleclanche.

**Comment by jleclanche** [comment]
Landed

**Pull request closed without merging**